automl/automl-script/readme.md

```diff
@@ -42,9 +42,29 @@ The **inputs** for the script are:
   `automl-execution` should be provided for the script to work. When
   both are provided, the `automl-execution` argument is discarded and
   new resources are generated from the training data.
-* `excluded-fields`: (list) List of fields that will be excluded from
-  any dataset before any other process starts. e.g. ["bmi",
-  "age"]. Empty by default
+* `configuration-params`: (map) Execution configuration
+  parameters. They will be overwritten if automl-execution is
+  given. Overwritten if automl-execution is given. The following
+  values must be provided:
+  * `excluded-fields`: (list) List of fields that will be excluded
+  from any dataset before any other process starts. e.g. ["bmi",
+  "age"]. Empty by default.
+  * `excluded-models`: (list) List of unsupervised models that won't
+  be created nor reused during feature generation. e.g. ["anomaly",
+  "cluster"]. Possible values are: association, cluster, anomaly, pca
+  or topicmodel. Empty by default
+  * `pca-variance-threshold`: (number) The PCA projection uses the
+  minimum number of components such that the cumulative explained
+  variance is greater than the given threshold. Values from 0 to 1.
+  Default value is 1 (all the components will be used)
+  * `leverage-threshold`: (number) Associations with an absolute
+  leverage lower than the threshold will be ignored.  Default value is
+  0 (all the association will be added to the extended dataset)
+
+
+**WARNING** Remember that, to avoid confusion, `configuration-params`
```

Comment:
I understand that excluded-fields will  not be used when the automl-execution is given, but the rest of parameters can be used, right?

Reply:
Not in my implementation. **All** the configuration params are inherited from the execution, if it is given. I  have thought about this and I took this decision because I think that it would be difficult to the user to distinguish between parameters that are inherited and parameter that are not. If we take the full automl as a new kind of model (as it could be a deepnet or a single optiml), these are parameters for **training** the model. If we pass a previous automl-execution is because we don't want to train, we only want to evaluate or predict, so we shouldn't change the configuration of the model without training it again. In fact, in this case, even the OptiML is inherated from the given automl-execution. If we change the configuration, e.g. of the execluded-models, and don't train a new OptiML we will have a lot of missing inputs in our prediction

Reply:
I would like to know your opinion about this

Reply:
I was thinking that if I have an existing automl-execution I might want to change the pca-variance-threshold or the leverage-threhsold and run another one. These are like "batch-prediction" config parameters, and can be reset using the same extended dataset and models.

Reply:
After our discussion, I see clearly that you are only using the execution as pointer to the final fusion model to create batch predictions based on the models as they have been built. In that sense, it's totally correct to assume that these parameters will not be used in the execution is provided.